src/main/java/luffy/Parser.java

```diff
@@ -0,0 +1,118 @@
+package luffy;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Parser class to handle user inputs.
+ *
+ * @author Silas Tay (A0233425M)
+ */
+public class Parser {
+    private Ui ui;
+
+    /**
+     * Constructor for Parser class.
+     */
+    public Parser() {
+        this.ui = new Ui();
+    }
+
+    /**
+     * Parses through user inputs and takes action based on command.
+     *
+     * @param s User input line
+     * @param tasks Tasklist of Luffy instance
+     */
+    public void parse(String s, TaskList tasks) {
```

Comment:
This `parse` method seems to form the main bulk of the logic that handles the user's command. Perhaps you can think of creating `Command` classes and extract the logic into each corresponding `Command` class. E.g. `DeadlineCommand` class.